### DISCUSSION TITLE
Random error NIV2_FLOPS

### DISCUSSION THREAD

**Discussion by srinath-chakravarthy** [idaholab/moose#15910] [Q&A General] [answered]
Hi all, 
I am having some trouble with simulations randomly stopping due to NIV2_FLOPS error. 

The problem is a contact problem with AD and i am using superlu_dist with Newton solver. After a random period of time, the simulation aborts with "problem with NIV2_FLOPS message  -5165450.0000000000                0  -5165450.0000000000". In the current avatar i have used 16 cores and there does not seem to be any memory issues for this problem. 

Any help will be appreciated. 

Cheers
Srinath

**Comment by lindsayad** [comment]
I'm very happy to see you using this forum @srinath-chakravarthy. It is going to take me a little bit to get my workflow setup so that I see discussions quickly. Right now you have to watch the whole repository to get new discussions posts notifications, and if I watch the whole repo I'm going to have to setup a filter because I don't care about every new issue and PR that gets opened.

I have never seen this error before. Some googling shows some hits for other FEM codes (I saw Elmer) and also a PETSc post. Most of these seem to be concerned with MUMPS, although apparently they submitted a [fix for that](https://scm.cs.kuleuven.be/scm/svn/numerics_software/MUMPS_5.0.1/ChangeLog). There was also this [PETSc thread](https://lists.mcs.anl.gov/pipermail/petsc-users/2014-March/020776.html). It's unclear to me from the message whether this is a CPU or memory error. What is the "current avatar"?

Perhaps @fdkong has some ideas about this

**Comment by fdkong** [comment]
What is the problem size? If the problem is large, it would be an out of integer error.

**Comment by srinath-chakravarthy** [comment, marked as answer]
Alex and fande,
Problem size is relatively small, while I'm not in front of my computer
now,  it has about 3000 quad elements, 3600 nodes with 5 dof per node. I
checked mem usage and it is small 126 MB for each processor with replicated
mesh. I have not faced this issue before... I'm more than happy to share my
repo with you as well as the input file.

Alex, I also Googled before posting and found the same things you did.

Cheers
Srinath
P.S: I just thought of giving discussions a try since it is being touted as
an alternative to the moose users forum. I am not sure about  workflow on
my side either ...

**Reply by fdkong** [reply]
It is interesting. Are you using a direct solver, eg, LU?

I was wondering what will happen if you try `-pc_factor_mat_solver_type mumps` and  `-pc_factor_mat_solver_type superlu_dist`, respectively? The reason is that we want to know where the error message comes from.

If you could reproduce this using a moose example, it would be a big help.

**Reply by aeslaughter** [reply]
I am marking this as the answer as the discussion seems to be stale. @srinath-chakravarthy If this is not the case, you can un-mark and continue the discussion.

**Comment by lindsayad** [comment]
He is using superlu_dist
